bidadjustment/build_rules.go

```diff
@@ -0,0 +1,93 @@
+package bidadjustment
+
+import (
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const (
+	VideoInstream  = "video-instream"
+	VideoOutstream = "video-outstream"
+)
+
+// BuildRules() will populate the rules map with a rule that's a combination of the mediaType, bidderName, and dealId for a particular adjustment
+// The result will be a map that'll map a given rule with its adjustment
+func BuildRules(bidAdjustments *openrtb_ext.ExtRequestPrebidBidAdjustments, rules map[string][]openrtb_ext.Adjustment) {
+	if bidAdjustments == nil {
+		return
+	}
+	buildRulesForMediaType(string(openrtb_ext.BidTypeBanner), bidAdjustments.MediaType.Banner, rules)
+	buildRulesForMediaType(string(openrtb_ext.BidTypeAudio), bidAdjustments.MediaType.Audio, rules)
+	buildRulesForMediaType(string(openrtb_ext.BidTypeNative), bidAdjustments.MediaType.Native, rules)
+	buildRulesForMediaType(VideoInstream, bidAdjustments.MediaType.VideoInstream, rules)
+	buildRulesForMediaType(VideoOutstream, bidAdjustments.MediaType.VideoOutstream, rules)
+	buildRulesForMediaType(WildCard, bidAdjustments.MediaType.WildCard, rules)
+}
+
+func buildRulesForMediaType(mediaType string, rulesByBidder map[openrtb_ext.BidderName]openrtb_ext.AdjustmentsByDealID, rules map[string][]openrtb_ext.Adjustment) {
+	for bidderName := range rulesByBidder {
+		for dealID, adjustments := range rulesByBidder[bidderName] {
+			rule := mediaType + Delimiter + string(bidderName) + Delimiter + dealID
+			rules[rule] = adjustments
+		}
+	}
+}
+
+func Merge(req *openrtb_ext.RequestWrapper, acctBidAdjs *openrtb_ext.ExtRequestPrebidBidAdjustments) (*openrtb_ext.ExtRequestPrebidBidAdjustments, error) {
+	mergedBidAdj, err := merge(req, acctBidAdjs)
+	if err != nil {
+		return nil, err
+	}
+	if !Validate(mergedBidAdj) {
```

Comment:
Discussed offline. We will stick with a warning but will need to adjust the warning message to indicate the issue is in the account instead of the request. We'll also need to append the error to the errors slice if the error returned by Merge is non-fatal.